### ISSUE TITLE
The default `synchronous` setting for sqlite auth backends should be something other than `OFF`

### ISSUE THREAD

**Issue by espadolini** [gravitational/teleport#11119]
### What
The current behavior for all sqlite backends (including process storage and auth backend) is to set sqlite's [`synchronous` pragma](https://www.sqlite.org/pragma.html#pragma_synchronous) to `OFF`.

It should be set to `FULL` when using deferred journal mode (the default) or `NORMAL` when using Write-Ahead Logging mode, after checking that the loss in performance doesn't meaningfully affect things.

### Why
With `synchronous` set to `OFF` a system crash or hard poweroff at the wrong time can corrupt the contents of the database.

### Workaround
It's possible to configure the `synchronous` pragma in the backend config file with the `sync` parameter, but it's not documented - we could just add some documentation about it, and after #10706 it will be possible to configure the `journal_mode` pragma as well, but what's a good justification for defaulting to `OFF`?

**Comment by espadolini** [comment]
As per @zmb3, we're going to change the default to `FULL`.
